libnetwork/endpoint_store_test.go

```diff
@@ -25,11 +26,22 @@ func TestEndpointStore(t *testing.T) {
 	err = c.storeEndpoint(context.Background(), ep2)
 	assert.NilError(t, err)
 
+	epSorter := func(a, b *Endpoint) int {
+		if a.id < b.id {
+			return -1
+		}
+		if a.id > b.id {
+			return 1
+		}
+		return 0
+	}
+
 	// Check that we can find both endpoints
 	found := c.findEndpoints(filterEndpointByNetworkId("testNetwork"))
+	slices.SortFunc(found, epSorter)
 	assert.Equal(t, len(found), 2)
-	assert.Equal(t, found[0], ep1)
-	assert.Equal(t, found[1], ep2)
+	assert.Check(t, found[0] == ep1, "got: %s; expected: %s", found[0].id, ep1.id)
+	assert.Check(t, found[1] == ep2, "got: %s; expected: %s", found[1].id, ep2.id)
```

Comment:
using `is.Equal` saves you from writing the message :) 
```suggestion
	assert.Check(t, is.Equal(found[0], ep1))
	assert.Check(t, is.Equal(found[1], ep2))
```

Reply:
`is.Equal` is probably more idiomatic but unfortunately it's going to dump the whole struct which isn't particularly readable. So I'll switch to it, but keep the custom message.

I could compare the endpoint ID instead, but one of the goal of this test is to make sure `findEndpoints` returns the original values (and not copy). I'll add a comment to clarify that.

Reply:
> one of the goal of this test is to make sure findEndpoints returns the original values (and not copy)

The comment says "and that the returned values are not copies of the original ones" - which seems like the opposite at first glance.

So, is the intention to check the pointers are the same (there's a single `Endpoint` object), rather than being pointers to different `Endpoint` objects that have the same values?

I think `is.Equal` will compare the pointed-to values if the pointers are different. So, it probably needs to go back to the original pointer equality?

(But then, if the pointers are to different objects with the same `id` value ... the error message will be confusing because the two ids will be the same?)

Reply:
I'm talking nonsense, it's not `DeepEqual` - ignore me!

Reply:
(Although, the printed `id` values are likely to be the same even if the pointers are different?)